### PR TITLE
f-checkout@v0.112.0 - Added validation on max length of api bound fields

### PR DESCRIPTION
**Pull request by oliversweb**
v0.112.0
------------------------------
*May 14, 2021*

### Added
- Max Length added to fields submitted to Api to match spec.

### CODE REVIEW

**Review by xander-marjoram — DISMISSED** [review]

File: packages/components/organisms/f-checkout/test/component/f-checkout-guest.component.desktop.spec.js [hunk 1]
```diff
@@ -29,17 +29,20 @@ describe('f-checkout "guest" component tests - @browserstack', () => {
         expect(pathname).toEqual(loginPath);
     });
 
-    it('should prevent a user from writing a note of over 200 characters', () => {
+    forEach([
+        ['firstName', 100],
+        ['lastName', 100],
+        ['userNote', 200]
+    ])
+    .it('should prevent a user from entering more characters than allowed in a field', (field, maxlength) => {
```
Comment (simonsandersje) [inline packages/components/organisms/f-checkout/test/component/f-checkout-guest.component.desktop.spec.js]: Could set the description as:
`should prevent a user from entering a "%s" with more than "%s" characters`

File: packages/components/organisms/f-checkout/test/component/f-checkout.component.shared.spec.js [hunk 2]
```diff
@@ -33,18 +36,24 @@ describe('f-checkout component tests - @browserstack', () => {
...
-    it('should prevent a user from writing a note of over 200 characters', () => {
+    forEach([
+        ['addressLine1', 255],
+        ['addressLine1', 255],
+        ['addressLocality', 50],
+        ['addressPostcode', 50],
+        ['mobileNumber', 16],
+        ['userNote', 200]
+    ])
+    .it('should prevent a user from entering more characters than allowed in a field', (field, maxlength) => {
```
Comment (simonsandersje) [inline packages/components/organisms/f-checkout/test/component/f-checkout.component.shared.spec.js]: As above

File: packages/components/organisms/f-checkout/src/components/Address.vue [hunk 3]
```diff
@@ -9,6 +9,7 @@
                 :value="address.line1"
                 :class="$style['c-address-formField']"
                 name="address-line-1"
+                maxlength="255"
```
Comment (1itvinka) [inline packages/components/organisms/f-checkout/src/components/Address.vue:12]: might be worth moving it to a reusable variable?

File: packages/components/organisms/f-checkout/test/component/f-checkout-guest.component.desktop.spec.js [hunk 4]
```diff
@@ -29,17 +29,20 @@ describe('f-checkout "guest" component tests - @browserstack', () => {
...
-        const userNote = 'A';
-        const addressInfo = {
-            note: userNote.repeat(300)
-        };
+        checkout.clearCheckoutForm(field);
+        const userEntry = 'A'.repeat(maxlength + 1); // Enter more than allowed
 
         // Act
-        checkout.inputUserNote(addressInfo);
+        checkout.setField(field, userEntry);
```
Comment (siggerzz) [inline packages/components/organisms/f-checkout/test/component/f-checkout-guest.component.desktop.spec.js]: Nit: Personally, I'd prefer `setFieldValue`

File: packages/components/organisms/f-checkout/test/component/f-checkout-guest.component.desktop.spec.js [hunk 5]
```diff
@@ -29,17 +29,20 @@ describe('f-checkout "guest" component tests - @browserstack', () => {
...
 
         // Assert
-        expect(checkout.userNoteMaxCharacterCount()).toEqual('200');
+        expect(checkout.getField(field).length).toEqual(maxlength);
```
Comment (siggerzz) [inline packages/components/organisms/f-checkout/test/component/f-checkout-guest.component.desktop.spec.js]: Nit: Personally, I'd prefer `getFieldValue`

**Review by siggerzz — DISMISSED** [review]

File: packages/components/organisms/f-checkout/src/components/Address.vue (hunk 3 above)
Comment (oliversweb) [reply packages/components/organisms/f-checkout/src/components/Address.vue:12]: Unsure the benefit can you expand on this?

File: packages/components/organisms/f-checkout/src/components/Checkout.vue [hunk 6]
```diff
@@ -839,6 +839,13 @@ export default {
                     this.shouldShowSpinner = true;
                 }
             }, this.spinnerTimeout);
+        },
+
+        formFieldBlur (field) {
```
Comment (1itvinka) [inline packages/components/organisms/f-checkout/src/components/Checkout.vue]: minor: not ideal that this is duplication

**Review by 1itvinka — DISMISSED** [review]

**Review by kevinrodrigues — DISMISSED** [review]

File: packages/components/organisms/f-checkout/src/components/Checkout.vue (hunk 6 above)
Comment (oliversweb) [reply packages/components/organisms/f-checkout/src/components/Checkout.vue]: Yes, the method name is the same in x3 files but the source of the data is different and then only a one-liner ... so more code to move in to a mixin than not ... so defeats the purpose of a mixin, so left as is.

**Review by kevinrodrigues — DISMISSED** [review]

File: packages/components/organisms/f-checkout/test/component/f-checkout-guest.component.desktop.spec.js [hunk 7]
```diff
@@ -29,17 +29,20 @@ describe('f-checkout "guest" component tests - @browserstack', () => {
         expect(pathname).toEqual(loginPath);
     });
 
-    it('should prevent a user from writing a note of over 200 characters', () => {
+    forEach([
+        [100, 'firstName'],
+        [100, 'lastName'],
+        [200, 'userNote']
```
Comment (SachaZvetelman) [inline packages/components/organisms/f-checkout/test/component/f-checkout-guest.component.desktop.spec.js]: I think this one is covered by the test in `shared`.

**Review by JoshuaNg2332 — APPROVED** [review]

**Review by 1itvinka — APPROVED** [review]

**Pull request merged**